src/weight_windows.cpp

```diff
@@ -815,13 +787,59 @@ WeightWindowsGenerator::WeightWindowsGenerator(pugi::xml_node node)
   // create a matching weight windows object
   auto wws = WeightWindows::create();
   ww_idx_ = wws->index();
+  wws->set_mesh(mesh_idx);
   if (e_bounds.size() > 0)
     wws->set_energy_bounds(e_bounds);
-  wws->set_mesh(model::mesh_map[mesh_id]);
   wws->set_particle_type(particle_type);
   wws->set_defaults();
 }
 
+void WeightWindowsGenerator::create_tally()
+{
+  const auto& wws = variance_reduction::weight_windows[ww_idx_];
+
+  // create a tally based on the WWG information
+  Tally* ww_tally = Tally::create();
+  tally_idx_ = model::tally_map[ww_tally->id()];
+  ww_tally->set_scores({"flux"});
+
+  int32_t mesh_id = wws->mesh()->id();
+  int32_t mesh_idx = model::mesh_map.at(mesh_id);
+  // see if there's already a mesh filter using this mesh
+  bool found_mesh_filter = false;
+  for (const auto& f : model::tally_filters) {
+    if (f->type() == FilterType::MESH) {
+      const auto* mesh_filter = dynamic_cast<MeshFilter*>(f.get());
+      if (mesh_filter->mesh() == mesh_idx && !mesh_filter->translated()) {
+        ww_tally->add_filter(f.get());
+        found_mesh_filter = true;
+        break;
+      }
+    }
+  }
+
+  if (!found_mesh_filter) {
+    auto mesh_filter = Filter::create("mesh");
+    openmc_mesh_filter_set_mesh(mesh_filter->index(), model::mesh_map[mesh_id]);
+    ww_tally->add_filter(mesh_filter);
+  }
+
+  const auto& e_bounds = wws->energy_bounds();
```

Comment:
we might want to do the same check as the mesh above to see if an `EnergyFilter` with the same bins or ID is used in another tally that has been added already.

Reply:
Yeah, I considered that. The duplicate mesh filters are going to be more costly than duplicate energy filters so I decided to focus on those instead.